flink-java/src/main/java/org/apache/flink/api/java/utils/ParameterTool.java

```diff
@@ -192,6 +192,7 @@ public static ParameterTool fromSystemProperties() {
 	 * @throws IOException If arguments cannot be parsed by {@link GenericOptionsParser}
 	 * @see GenericOptionsParser
 	 */
+	@Deprecated
```

Comment:
@greghogan  The `GenericOptionsParser` is a Hadoop utility, so it should not be directly referenced in `flink-java`